### ISSUE TITLE
Debian Jessie package: alert_new_files, report_changes, agent_control -r don't work

### ISSUE THREAD

**Issue by naturalspringwater** [ossec/ossec-hids#718]
I installed ossec-hids packages 2.8.3-3jessie from http://ossec.wazuh.com/repos/apt/debian.
Started with (and after every change)
systemctl restart ossec 

"alert_new_files" produces no alerts after adding to syscheck and adding the following to /var/ossec/rules/local_rules.xml:

```
  <rule id="554" level="10" overwrite="yes">
    <category>ossec</category>
    <decoded_as>syscheck_new_entry</decoded_as>
    <description>File added to the system.</description>
    <group>syscheck,</group>
  </rule>

</group> <!-- SYSLOG,LOCAL -->
```

"report_changes" produces no diffs by email nor in /var/ossec/logs/alerts/alerts.log. On one occasion it gave these strange errors in /var/ossec/logs/ossec.log (why is it trying to rename those files?):

```
2015/12/16 22:02:46 ossec-syscheckd: INFO: Starting syscheck scan (forwarding database).
2015/12/16 22:02:46 ossec-syscheckd: INFO: Starting syscheck database (pre-scan).
2015/12/16 22:02:46 ossec-syscheckd(1124): ERROR: Unable to rename file: '/etc/systemd/system/dbus-org.freedesktop.Avahi.service$
2015/12/16 22:02:50 ossec-syscheckd(1124): ERROR: Unable to rename file: '/etc/systemd/system/multi-user.target.wants/avahi-daem$
2015/12/16 22:02:52 ossec-syscheckd(1124): ERROR: Unable to rename file: '/etc/systemd/system/sockets.target.wants/avahi-daemon.$
2015/12/16 22:16:56 ossec-syscheckd: INFO: Finished creating syscheck database (pre-scan completed).
2015/12/16 22:17:08 ossec-syscheckd: INFO: Ending syscheck scan (forwarding database).
```

The command

```
/var/ossec/bin/agent_control -r -u 000
```

has no effect and appears to be ignored by syscheckd (no reaction despite the command line saying "OSSEC HIDS agent_control: Restarting Syscheck/Rootcheck locally").

My config:

```
  <syscheck>
    <!-- Frequency that syscheck is executed -- default every 20 hours -->
    <frequency>72000</frequency>

    <!-- Directories to check  (perform all possible verifications) -->
    <directories report_changes="yes" check_all="yes">/etc,/usr/bin,/usr/sbin</directories>
    <directories report_changes="yes" check_all="yes">/bin,/sbin</directories>

    <!-- Files/directories to ignore -->
    <ignore>/etc/mtab</ignore>
    <ignore>/etc/hosts.deny</ignore>
    <ignore>/etc/mail/statistics</ignore>
    <ignore>/etc/random-seed</ignore>
    <ignore>/etc/adjtime</ignore>
    <ignore>/etc/httpd/logs</ignore>
    <ignore>/etc/test.test2</ignore>

    <alert_new_files>yes</alert_new_files>
    <scan_on_start>yes</scan_on_start>
    <auto_ignore>no</auto_ignore>

  </syscheck>
```

ossec appears to scan logs successfully and sends emails on basic log changes and server status and on file changes when the scan does run (but not on new files and no diffs and can't trigger scan manually). The config file ossec.conf does clearly get read (lists any extra ignores I add in log).

I'm at a loss as to why none of these are working, and the syscheck/rootcheck scans each take 20mins which makes debugging a nightmare.

**Comment by naturalspringwater** [comment]
After multiple reboots, the command 
[code]
/var/ossec/bin/agent_control -r -u 000
[/code]
now appears to be responding, but with a 10min delay before any "Starting..." log entries show up.

**Comment by naturalspringwater** [comment]
Is there any way to run syscheck/rootcheck without any throttling, for debugging??

**Comment by naturalspringwater** [comment]
After fighting a whole more day with it, clearing database with syscheck_update over and over again, etc., I finally got it to send diffs and alerts for new files. I had to disable scan_on_start (default is enabled), with it enabled the results appear unpredictable.

I still ask, is there any way to disable the throttling and delays? Because it's a nightmare to debug with.

**Comment by lostinthetubez** [comment]
Looks like there are a couple performance options in /var/ossec/etc/internal_options.conf. Open it up and search for syscheck.

You might also want to ask questions like these on the OSSEC mailing list instead of the development forums. You may (or may not) get a better response. Just another resource if you're having trouble.

https://groups.google.com/forum/#!forum/ossec-list

**Comment by naturalspringwater** [comment]
Thanks. I'd skipped internal_options.conf file because its header says "DO NOT TOUCH THIS FILE".

I set syscheck.sleep=0 inside and now syscheck only takes a minute. However it doesn't affect rootcheck, which still takes 20 minutes. I checked the source and it looks as though the sleep for rootcheck is hardcoded.

It also doesn't remove the the 5-10min delay before the check starts after sending a manual restart with agent_control.

I will probably email later.

As for this ticket, there was definitely buggy behavior out of the box but it's entered the realm where I can't reproduce it consistently, though this is with scan_on_start disabled, which I won't reenable.
